app/controllers/users_controller.rb

```diff
@@ -0,0 +1,37 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new(username: 'Test',
+                     email: 'test@test.com',
+                     password: 'omaromar')
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      redirect_to new_user_path
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update(username: params[:user]['username'], password: params[:user]['password'],
+                    email: params[:user]['email'])
```

Comment:
- [ ] [required] Since you have `user_params` in your private methods, kindly use them instead of picking users submitted params directly. This way could be better than the one you used: `@user.update(user_params)` instead of ~`@user.update(username: params[:user]['username'], password: params[:user]['password'],email: params[:user]['email'])`~ Because you need strong parameters for the security purposes